src/dmd/lambdacomp.d

```diff
@@ -435,4 +440,33 @@ public:
         else
             visitType(p.type);
     }
-}
+
+    override void visit(ArrayLiteralExp) { buf.reset(); }
+    override void visit(AssocArrayLiteralExp) { buf.reset(); }
+    override void visit(CompileExp) { buf.reset(); }
+    override void visit(ComplexExp) { buf.reset(); }
+    override void visit(DeclarationExp) { buf.reset(); }
+    override void visit(DefaultInitExp) { buf.reset(); }
+    override void visit(DsymbolExp) { buf.reset(); }
+    override void visit(ErrorExp) { buf.reset(); }
+    override void visit(FuncExp) { buf.reset(); }
+    override void visit(HaltExp) { buf.reset(); }
+    override void visit(IntervalExp) { buf.reset(); }
+    override void visit(IsExp) { buf.reset(); }
+    override void visit(NewAnonClassExp) { buf.reset(); }
+    override void visit(NewExp) { buf.reset(); }
+    override void visit(NullExp) { buf.reset(); }
+    override void visit(ObjcClassReferenceExp) { buf.reset(); }
+    override void visit(OverExp) { buf.reset(); }
+    override void visit(ScopeExp) { buf.reset(); }
+    override void visit(StringExp) { buf.reset(); }
+    override void visit(StructLiteralExp) { buf.reset(); }
+    override void visit(SymbolExp) { buf.reset(); }
+    override void visit(TemplateExp) { buf.reset(); }
+    override void visit(ThisExp) { buf.reset(); }
+    override void visit(TraitsExp) { buf.reset(); }
+    override void visit(TupleExp) { buf.reset(); }
+    override void visit(TypeExp) { buf.reset(); }
+    override void visit(TypeidExp) { buf.reset(); }
+    override void visit(VoidInitExp) { buf.reset(); }
+}
```

Comment:
Wouldn't simply overwriting the Expression node account for all those cases? 

```d
override void visit(Expression) { buf.reset(); }
```

Reply:
That seems reasonable, and was the first thing I tried, but I can't seem to get it to work.